packages/elements/src/chart/plugins/doughnut-center-label.ts

```diff
@@ -1,257 +1,275 @@
-//         const numText = targetText.length;
-//         const numRatio = targetText.length / ctx.measureText(targetText).width;
-//         const difSize = ctx.measureText(targetText).width - containerWidth + circlePadding;
-//         let numCharacter = numText - Math.ceil(difSize * numRatio);
-//         numCharacter = Math.max(1, numCharacter);
-//         if (numCharacter) {
-//           targetText = targetText.slice(0, numCharacter);
-//           targetText += '...';
-//         }
-//       }
-//       ctx.fillText(targetText, centerX, offsetY + (i * fontLineHeight));
-//     }
-//   },
-//   afterDatasetUpdate: function (chart: DoughnutChart & Selectable): void {
-
-//     // Check chart already init selected.
-//     if (chart.selected !== undefined) {
-//       return;
-//     }
-
-//     // Check configs are ready to init selected
-//     const config = getPluginConfig(chart);
-//     if (!config || !config.selected || !chart.data.datasets) {
-//       return;
-//     }
-
-//     const selectedIndex = Number(config.selected?.index);
-//     const datasetIndex = Number(config.selected.datasetIndex || 0);
-//     const visibleIndexes = chart.getVisibleDatasetCount() - 1;
-
-//     // Validate index and datasetIndex
-//     if (isNaN(selectedIndex) || (isNaN(datasetIndex) || datasetIndex < 0 || datasetIndex > visibleIndexes)) {
-//       return;
-//     }
-
-//     // Set selected item to the chart
-//     const items = chart.getDatasetMeta(datasetIndex).data[selectedIndex];
-//     chart.selected = items ? [items] : [];
-//   },
-//   afterDatasetsDraw: function (chart: DoughnutChart & Selectable): void {
-//     if (getPluginConfig(chart)) {
-//       // Draw active element
-//       // Note: use logic from chart.js - chart.js/src/elements/element.arc.js :draw()
-
-//       // hover
-//       drawItemBorder(chart, chart.active);
-
-//       // selected
-//       if (chart.selected?.length) {
-//         chart._select = chart.selected;
-//         chart.selected = [];
-//       }
-
-//       // active
-//       drawItemBorder(chart, chart._select);
-//     }
-//   }
-// };
-
-// export default plugins;
+import {
+  Chart as ChartJS,
+  Plugin,
+  ChartEvent,
+  PluginOptionsByType,
+  ChartType,
+  DoughnutController,
+  ActiveElement,
+  ArcOptions,
+  ArcElement
+} from 'chart.js';
+import { color as parseColor } from '@refinitiv-ui/utils/color.js';
+
+interface Selectable {
+  _select: ActiveElement[];
+  selected: ActiveElement[];
+  active: ActiveElement[];
+}
+
+interface CenterLabel {
+  label: string;
+  bold?: boolean;
+}
+
+interface Selected {
+  datasetIndex?: number;
+  index: number;
+}
+
+interface CenterLabelConfig {
+  defaultText: CenterLabel[];
+  selected: Selected;
+  onRenderLabel(chart: ChartJS, chartItems: ActiveElement[]): CenterLabel[];
+}
+
+declare module 'chart.js' {
+  interface PluginOptionsByType<TType extends ChartType> {
+    'centerLabel'?: CenterLabelConfig
```

Comment:
As we discussed, we aren't going to change. Because if we change to be kebab-case, it will cause breaking change that users have to put id plugin as key string. It will be improper to declare object like below:
```javascript
options: {
 plugins: {
   'center-label': {
     defaultText: ...,
     ...
   }
 }
}
```